pkg/entrypoint/run.go

```diff
@@ -0,0 +1,295 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package entrypoint
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// InternalErrorCode is what we write to the marker file to
+	// indicate that we failed to start the wrapped command
+	InternalErrorCode = 127
+	// AbortedErrorCode is what we write to the marker file to
+	// indicate that we were terminated via a signal.
+	AbortedErrorCode = 130
+
+	// DefaultTimeout is the default timeout for the test
+	// process before SIGINT is sent
+	DefaultTimeout = 120 * time.Minute
+
+	// DefaultGracePeriod is the default timeout for the test
```

Comment:
Probably another Prowism, what's "the test process" here?

Reply:
makes sense to remove, I don't believe the graceful termination logic is too useful here.  The test process here is the process that the entrypoint binary will run (the original entrypoint/command for the container).